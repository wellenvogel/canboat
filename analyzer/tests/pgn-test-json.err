INFO pgn-test [analyzer] Timestamp fixed
INFO pgn-test [analyzer] Detected FAST format with all frames on one line
ERROR pgn-test [analyzer] PGN 129540 has 2 missing fields in repeating set
