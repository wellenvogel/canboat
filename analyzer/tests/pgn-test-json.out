{"timestamp":"2011-04-25-06:25:03.603","prio":3,"src":36,"dst":255,"pgn":129029,"description":"GNSS Position Data","fields":{"SID":230,"Date":"2011.04.25","Time":"06:25:12","Latitude":52.7461333,"Longitude": 5.1815566,"Altitude":3.400000,"GNSS type":"GPS+SBAS/WAAS","Method":"GNSS fix","Integrity":"No integrity checking","Number of SVs":9,"HDOP":0.90,"PDOP":1.40,"Reference Stations":0}}
{"timestamp":"1970-01-01T00:00:00.000Z","prio":3,"src":61,"dst":255,"pgn":127513,"description":"Battery Configuration Status","fields":{"Instance":0,"Battery Type":"Gel","Supports Equalization":"No","Nominal Voltage":"12V","Chemistry":"Li","Capacity":20,"Temperature Coefficient":2,"Peukert Exponent":0.002,"Charge Efficiency Factor":98}}
{"timestamp":"2022-09-10T12:10:16.614Z","prio":6,"src":5,"dst":255,"pgn":60928,"description":"ISO Address Claim","fields":{"Unique Number":1088507,"Manufacturer Code":"Navico","Device Instance Lower":0,"Device Instance Upper":0,"Device Function":"Rudder","Device Class":"Steering and Control surfaces","System Instance":0,"Industry Group":"Marine","Arbitrary address capable":1}}
{"timestamp":"2022-09-10T12:10:16.812Z","prio":6,"src":35,"dst":255,"pgn":60928,"description":"ISO Address Claim","fields":{"Unique Number":321561,"Manufacturer Code":"Airmar","Device Instance Lower":0,"Device Instance Upper":0,"Device Function":"Bottom Depth","Device Class":"Navigation","System Instance":0,"Industry Group":"Marine","Arbitrary address capable":1}}
{"timestamp":"2016-04-09T16:41:39.628Z","prio":2,"src":16,"dst":255,"pgn":127489,"description":"Engine Parameters, Dynamic","fields":{"Instance":"Single Engine or Dual Engine Port","Oil pressure":1.583,"Temperature":23.52,"Alternator Potential":13.81,"Total Engine hours":"01:10:10","Discrete Status 1":["Over Temperature","Low Oil Pressure"]}}
{"timestamp":"1970-01-01T16:41:39.628Z","prio":2,"src":16,"dst":255,"pgn":127489,"description":"Engine Parameters, Dynamic","fields":{"Instance":"Single Engine or Dual Engine Port","Oil pressure":1.583,"Oil temperature":547.65,"Temperature":23.52,"Alternator Potential":13.81,"Fuel Rate":112.5,"Total Engine hours":"01:10:10","Coolant Pressure":8.208,"Fuel Pressure":164.320,"Discrete Status 1":["Over Temperature","Low Oil Pressure"],"Discrete Status 2":["Warning Level 1","Warning Level 2","Power Reduction","Maintenance Needed","Engine Comm Error","Sub or Secondary Throttle","Neutral Start Protect","Engine Shutting Down"],"Engine Load":48,"Engine Torque":24}}
{"timestamp":"2022-09-10T12:10:33.618Z","prio":6,"src":23,"dst":255,"pgn":129540,"description":"GNSS Sats in View","fields":{"SID":4,"Sats in View":18,"list":[{"PRN":3,"Elevation":52.0,"Azimuth":88.0,"SNR":33.00,"Range residuals":0,"Status":"Used"},{"PRN":87,"Elevation":47.0,"Azimuth":191.0,"SNR":33.00,"Range residuals":0,"Status":"Used"},{"PRN":4,"Elevation":74.0,"Azimuth":144.0,"SNR":32.00,"Range residuals":0,"Status":"Used"},{"PRN":72,"Elevation":67.0,"Azimuth":39.0,"SNR":32.00,"Range residuals":0,"Status":"Used"},{"PRN":73,"Elevation":21.0,"Azimuth":53.0,"SNR":32.00,"Range residuals":0,"Status":"Used"},{"PRN":49,"Elevation":29.0,"Azimuth":181.0,"SNR":31.00,"Range residuals":0,"Status":"Used"},{"PRN":88,"Elevation":62.0,"Azimuth":286.0,"SNR":31.00,"Range residuals":0,"Status":"Used"},{"PRN":6,"Elevation":41.0,"Azimuth":304.0,"SNR":30.00,"Range residuals":0,"Status":"Used"},{"PRN":81,"Elevation":18.0,"Azimuth":331.0,"SNR":30.00,"Range residuals":0,"Status":"Used"},{"PRN":9,"Elevation":46.0,"Azimuth":210.0,"SNR":29.00,"Range residuals":0,"Status":"Used"},{"PRN":17,"Elevation":26.0,"Azimuth":228.0,"SNR":29.00,"Range residuals":0,"Status":"Used"},{"PRN":19,"Elevation":36.0,"Azimuth":258.0,"SNR":29.00,"Range residuals":0,"Status":"Used"},{"PRN":71,"Elevation":18.0,"Azimuth":57.0,"SNR":28.00,"Range residuals":0,"Status":"Used"},{"PRN":65,"Elevation":54.0,"Azimuth":256.0,"SNR":27.00,"Range residuals":0,"Status":"Used"},{"PRN":11,"Elevation":9.0,"Azimuth":312.0,"SNR":26.00,"Range residuals":0,"Status":"Used"},{"PRN":1,"Elevation":20.0,"Azimuth":147.0,"SNR":23.00,"Range residuals":0,"Status":"Used"},{"PRN":25,"Elevation":4.0,"Azimuth":357.0,"SNR":22.00,"Range residuals":0,"Status":"Used"},{"PRN":74,"Elevation":11.0,"Azimuth":98.0,"SNR":21.00,"Range residuals":0}]}}
{"timestamp":"2020-08-22-13:52:57.591","prio":7,"src":36,"dst":255,"pgn":126993,"description":"Heartbeat","fields":{"Data transmit offset":"00:00:00.001","Sequence Counter":36}}
{"timestamp":"2011-04-25-10:16:40.505","prio":3,"src":36,"dst":255,"pgn":126992,"description":"System Time","fields":{"SID":16,"Source":"GPS","Date":"2011.04.25","Time":"10:16:50.0001"}}
{"timestamp":"2021-01-30-20:43:21.684","prio":6,"src":1,"dst":255,"pgn":126998,"description":"Configuration Information","fields":{"Installation Description #1":"hello","Installation Description #2":"w√≥rld"}}
{"timestamp":"2020-04-19T00:35:55.571Z","prio":2,"src":0,"dst":67,"pgn":126208,"description":"NMEA - Command group function","fields":{"Function Code":"1","PGN":126998,"Number of Parameters":1,"list":[{"Parameter":2,"Value":"YD:VOLUME 60"}]}}
{"timestamp":"2021-07-29T10:18:31.758Z","prio":6,"src":36,"dst":0,"pgn":126208,"description":"NMEA - Acknowledge group function","fields":{"Function Code":"2","PGN":65410,"PGN error code":"Acknowledge","Transmission interval/Priority error code":"Transmit Interval/Priority not supported","Number of Parameters":2,"list":[{"Parameter":"Acknowledge"},{"Parameter":"Acknowledge"}]}}
{"timestamp":"2021-07-29T10:18:31.758Z","prio":6,"src":36,"dst":0,"pgn":126208,"description":"NMEA - Read Fields group function","fields":{"Function Code":"3","PGN":130306,"Unique ID":0,"Number of Selection Pairs":1,"Number of Parameters":2,"list":[{"Selection Parameter":4,"Selection Value":"Apparent"}],"list2":[{"Parameter":2},{"Parameter":3}]}}
{"timestamp":"2022-10-11T11:47:22Z","prio":3,"src":127,"dst":255,"pgn":126464,"description":"PGN List (Transmit and Receive)","fields":{"Function Code":"Receive PGN list","list":[{"PGN":130820},{"PGN":129809}]}}
{"timestamp":"2022-11-14T01:47:30.890Z","prio":2,"src":14,"dst":255,"pgn":127251,"description":"Rate of Turn","fields":{"Rate":-0.029649}}
{"timestamp":"2022-09-10T12:07:29.542Z","prio":4,"src":23,"dst":255,"pgn":129039,"description":"AIS Class B Position Report","fields":{"Message ID":"Standard Class B position report","Repeat Indicator":"Initial","User ID":"244180106","Longitude": 5.3134516,"Latitude":52.9061666,"Position Accuracy":"High","RAIM":"in use","Time Stamp":29,"COG":171.7,"SOG":1.80,"Communication State":"F8 08 00","AIS Transceiver information":"Channel A VDL reception","Unit type":"SOTDMA","Integrated Display":"No","DSC":"Yes","Band":"Entire marine band","Can handle Msg 22":"Yes","AIS mode":"Assigned","AIS communication state":"SOTDMA"}}
